docs/exposing-binding-data/olm-descriptors.md

```diff
@@ -0,0 +1,184 @@
+---
+sidebar_position: 5
+---
+
+# Adding Binding Metadata as OLM Descriptors
+
+If your operator is distributed as an [Operator
+Bundle](https://github.com/operator-framework/operator-registry/blob/master/docs/design/operator-bundle.md),
+you can add [OLM
+descriptors](https://github.com/openshift/console/blob/master/frontend/packages/operator-lifecycle-manager/src/components/descriptors/reference/reference.md)
+to describe what binding data are exposed.  The OLM descriptors are part of
+[Cluster Service
+Version](https://docs.openshift.com/container-platform/4.7/operators/operator_sdk/osdk-generating-csvs.html)
+files.  To expose the binding data, you can use SpecDescriptors and
+StatusDescriptors.  To specify a path under the `.spec` of a custom resource,
+use SpecDescriptors.  Similarly, to specify a path under `.status` of a custom
+resource, use StatusDescriptors.
+
+The secret generation behavior explained in the annotations is applicable for
+OLM descriptors also.  The data model is also same as that of annotations, but
+the syntax is different to match OLM descriptors naming pattern.
+
+The only two fields that is used for binding is `Path` and `X-Descriptors`.
+Path is a dot-delimited path of the field on the object that the descriptor
+describes. The X-Descriptors define the binding meta-data similar to CR/CRD
+annotation.
+
+If the path is pointing to a `Secret` resource, there should be an X-Descriptors like this:
+
+    urn:alm:descriptor:io.kubernetes:Secret
+
+Similary, if the path is pointing to a `ConfigMap` resource, there should be an X-Descriptors like this:
+
+    urn:alm:descriptor:io.kubernetes:ConfigMap
+
+If the `Secret` or `ConfigMap` specific X-Descriptors are not present, that
+descriptor should be accessing multiple values at the given path.
+
+There should be a `service.binding` entry in the X-Descriptors to identify this
+is a Service Binding configguration.
+
+## Mount an entire Secret as the binding Secret
+
+One of the common use case is to inject all the values from a Secret resource.
+The Secret resource must be specified as an attribute in the custom resource.
+If the attribute is part of `.spec`, you can create SpecDescriptor, and if it's
+part of the `.status`, you can create StatusDescriptor.  In the descriptor, you
+can use `path` attribute to point to field with the Secret resource name.
+
+Here is an example configuration:
+
+```
+- path: data.dbCredentials
+  x-descriptors:
+  - urn:alm:descriptor:io.kubernetes:Secret
+  - service.binding
+```
+
+The X-Descriptor has `urn:alm:descriptor:io.kubernetes:Secret` entry to indicate
+the path is pointing to a Secret resource.
+
+## Mount an entire ConfigMap as the binding Secret
+
+Another common use case is to inject all the values from a ConfigMap resource.
+The ConfigMap resource must be specified as an attribute in the custom resource.
+If the attribute is part of `.spec`, you can create SpecDescriptor, and if it's
```

Comment:
what are those SpecDescriptors and StatusDescriptors? I think it would help to reference that in the doc.